coderd/prometheusmetrics/aggregator.go

```diff
@@ -68,7 +67,32 @@ type annotatedMetric struct {
 var _ prometheus.Collector = new(MetricsAggregator)
 
 func (am *annotatedMetric) is(req updateRequest, m *agentproto.Stats_Metric) bool {
-	return am.username == req.username && am.workspaceName == req.workspaceName && am.agentName == req.agentName && am.Name == m.Name && slices.Equal(am.Labels, m.Labels)
+	return am.username == req.username &&
+		am.workspaceName == req.workspaceName &&
+		am.agentName == req.agentName &&
+		am.templateName == req.templateName &&
+		am.Name == m.Name &&
+		labelsEqual(am.Labels, m.Labels)
+}
+
+func labelsEqual(a, b []*agentproto.Stats_Metric_Label) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	am := make(map[string]string)
+	for _, l := range a {
+		am[l.GetName()] = l.GetValue()
+	}
+	for _, l := range b {
+		v, ok := am[l.GetName()]
+		if !ok {
+			return false
+		}
+		if v != l.GetValue() {
+			return false
+		}
```

Comment:
nit: alternatively, unless empty values for labels are appropriate.

```
if v := am[l.GetName()]; v != l.GetValue() {
    return false
}
```

Reply:
Just looked this up, and Prometheus treats empty label values as equivalent to not existing. To handle that case I'd need to rework the initial check on length.